src/kprofile.cpp

```diff
@@ -25,11 +28,24 @@ void KProfile::startDownload(QWebEngineDownloadRequest* download)
 #endif
 {
     auto app = KiwixApp::instance();
+    bool isHTML = false;
+    try
+    {
+        const auto item = getZimItem(download->url());
+        auto mimeType = QByteArray::fromStdString(item.getMimetype());
+        mimeType = mimeType;
+        isHTML = mimeType == "text/html";
+    }
+    catch (...) { /* Blank */ }
```

Comment:
For `try/catch`, the reason is download action is not only triggered by our Save_Page action. For example, if the user opens a video, there is a download button for the video. Though the downloads are rarely used, it would indeed be a bug.